src/main/java/org/dita/dost/module/XsltModule.java

```diff
@@ -169,40 +151,62 @@ private void transform(final File in, final File out) throws DITAOTException {
             logger.debug("Set parameter " + filedirparameter + " to '" + v + "'");
             t.setParameter(new QName(filedirparameter), new XdmAtomicValue(v));
         }
+
+        if (properties.isEmpty()) {
+            try {
+                if (same) {
+                    job.getStore().transform(in.toURI(), t);
+                } else {
+                    job.getStore().transform(in.toURI(), out.toURI(), t);
+                }
+            } catch (final UncheckedXPathException e) {
+                logger.error("Failed to transform document: " + e.getXPathException().getMessageAndLocation(), e);
+            } catch (final RuntimeException e) {
+                throw e;
+            } catch (final Exception e) {
+                logger.error("Failed to transform document: " + e.getMessage(), e);
+            }
+            return;
+        }
+
+        final File tmp = same ? new File(out.getAbsolutePath() + ".tmp" + Long.toString(System.currentTimeMillis())) : out;
         if (same) {
             logger.info("Processing " + in.getAbsolutePath());
             logger.debug("Processing " + in.getAbsolutePath() + " to " + tmp.getAbsolutePath());
         } else {
             logger.info("Processing " + in.getAbsolutePath() + " to " + tmp.getAbsolutePath());
         }
-        final Source source = new SAXSource(parser, new InputSource(in.toURI().toString()));
         try {
-            if (!tmp.getParentFile().exists() && !tmp.getParentFile().mkdirs()) {
-                throw new IOException("Failed to create directory " + tmp.getParent());
-            }
+            final Source source = job.getStore().getSource(in.toURI());
             t.setSource(source);
-            final Serializer serializer = processor.newSerializer(tmp);
-            for (final String key: properties.stringPropertyNames()) {
-                serializer.setOutputProperty(new QName(key), properties.getProperty(key));
+            final Destination destination = job.getStore().getDestination(tmp.toURI());
+            if (same) {
+                destination.setDestinationBaseURI(out.toURI());
+            }
+            if (destination instanceof Serializer) {
```

Comment:
Only set output properties if the destination is a Serializer.

`Destination` can be e.g. a tree builder that results in a S9api `XdmNode`, or then it can be something that serializes XML into a byte stream.